.github/workflows/draft_new_release.yml

```diff
@@ -9,32 +9,72 @@ jobs:
     name: "Draft a new release"
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
-        with:
-          ref: 'release'
       - name: Extract version from milestone
         run: |
           VERSION="${{ github.event.milestone.title }}"
-          echo "RELEASE_VERSION=${VERSION/v/}" >> $GITHUB_ENV
+          RELEASE_VERSION="${VERSION/v/}"
+          # Transforms 1.13.2 to 1.13 so that we can re-use the same
+          # branch for patch releases.
+          BRANCH_VERSION="${RELEASE_VERSION:$i:-2}"
+          if [[ "${RELEASE_VERSION: -1}" == "0" ]]; then
+            CHECKOUT_REF="develop"
+          else
+            CHECKOUT_REF="release-${BRANCH_VERSION}"
+          fi
+
+          # Export variables separately so the scripting above is more legible,
+          # and we can actually use them within this block. Changes to $GITHUB_ENV
+          # only affect the next step, not the current one.
+          echo "RELEASE_VERSION=${RELEASE_VERSION}" >> $GITHUB_ENV
+          echo "CHECKOUT_REF=${CHECKOUT_REF}" >> $GITHUB_ENV
+          echo "BRANCH_VERSION=${BRANCH_VERSION}" >> $GITHUB_ENV
+          echo "PR_BASE=release-${BRANCH_VERSION}" >> $GITHUB_ENV
+          echo "PR_HEAD=release-prep" >> $GITHUB_ENV
+
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          ref: ${{ env.CHECKOUT_REF }}
 
       - name: Update changelog
         uses: thomaseizinger/keep-a-changelog-new-release@96ebf19f2bddaf72406c84691b9c2d827da53140
         with:
-          version: ${{ env.RELEASE_VERSION }}
+          version: ${{ github.event.milestone.title }}
```

Comment:
We need the 'v' prefix here for changelog links to correctly work (tags have the prefix) so I simply switched it to the milestone title which has the prefix.